pageserver/src/tenant/timeline.rs

```diff
@@ -5550,6 +5550,9 @@ impl<'a> TimelineWriter<'a> {
         value: &Value,
         ctx: &RequestContext,
     ) -> anyhow::Result<()> {
+        if !key.is_i128_representable() {
+            bail!("the request contains data not supported by pageserver");
```

Comment:
What's the behavior on errors here: will this cause the wal ingest loop to continuously spin and retry ingesting the WAL?

(I agree with the underlying point that we should reject such data as early as possible)